examples/createAndInitPool/createAndInitPoolV3.ts

```diff
@@ -244,71 +228,45 @@ async function createPool({ client, call, userAccount }) {
     const {
         args: { pool: poolAddress },
     } = poolCreatedEvent;
+    console.log('Pool created at address: ', poolAddress);
+
     return poolAddress;
 }
 
-async function createPermit2({
+async function buildInitPoolCall({
     client,
-    userAccount,
-    initAmounts,
-    chainId,
+    rpcUrl,
+    account,
+    poolAddress,
 }: {
     client: PublicWalletClient;
-    userAccount: PrivateKeyAccount;
-    initAmounts: {
-        address: `0x${string}`;
-        rawAmount: bigint;
-        decimals: number;
-    }[];
-    chainId: ChainId;
+    rpcUrl: string;
+    account: Address;
+    poolAddress: Address;
 }) {
-    const owner = userAccount.address;
-    const permit2Contract = getContract({
-        address: PERMIT2[chainId],
-        abi: permit2Abi,
-        client,
-    });
-
-    const details: PermitDetails[] = await Promise.all(
-        initAmounts.map(async (token) => {
-            const [, , nonce] = await permit2Contract.read.allowance([
-                owner,
-                token.address,
-                BALANCER_ROUTER[chainId],
-            ]);
-
-            return {
-                token: token.address,
-                amount: token.rawAmount,
-                expiration: Number(MaxAllowanceExpiration),
-                nonce,
-            };
-        }),
+    // Fetch the necessary pool state
+    const initPoolDataProvider = new InitPoolDataProvider(chainId, rpcUrl);
+    const poolState = await initPoolDataProvider.getInitPoolData(
+        poolAddress,
+        poolType,
+        protocolVersion,
     );
 
-    const batch: Permit2Batch = {
-        details,
-        spender: BALANCER_ROUTER[chainId],
-        sigDeadline: MaxSigDeadline,
-    };
+    // Sign permit2 approval
+    const permit2 = await Permit2Helper.signInitPoolApproval({
+        ...initPoolInput,
+        client,
+        owner: account,
+    });
```

Comment:
`runAgainstNetwork` fails here because of an unsupported method

```
HttpRequestError: HTTP request failed.

Status: 400

Details: {"code":-32600,"message":"Unsupported method: eth_signTypedData_v4. }
```

Reply:
I confirmed that it is possible to modify the type for [owner](https://github.com/balancer/b-sdk/blob/74d61ab3476134b2406bc14f33e81b8e83752798/src/entities/permit2Helper/index.ts#L293) of `Permit2Helper.signPermit2` to be `Address | Account`  and that makes the script work if I provide an `Account`, but I'm assuming I am missing something about the client setup because I saw the v3 frontend repo is able to pass just an address and the `Permit2Helper` works there :thinking:

Reply:
Here is how I'm creating the client for reference
https://github.com/balancer/b-sdk/blob/c7a6c0d62fc064cd6dcb1a7054c9550cec4dd02b/examples/createAndInitPool/createAndInitPoolV3.ts#L162-L168

Reply:
Everything looks right on your client setup from what I can see.
Tracing into the viem code it seems like an Address is parsed to a `type: 'json-rpc'` which then branches to a client request:
```
client.request({
        method: 'eth_signTypedData_v4',
        params: [account.address, typedData],
    }
```
So it appears we have to use an `Account`